packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/services/NodeIncrementalLayoutEngineService.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.layout.services;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.diagrams.layout.incremental.INodeIncrementalLayoutEngine;
+import org.eclipse.sirius.components.diagrams.layout.incremental.INodeIncrementalLayoutEngineRegistry;
+
+/**
+ * Service used to get the first {@link INodeIncrementalLayoutEngine} available for the node type.
+ *
+ * @author gcoutable
+ */
+public class NodeIncrementalLayoutEngineService implements INodeIncrementalLayoutEngineService {
+
+    private final INodeIncrementalLayoutEngineRegistry registry;
+
+    public NodeIncrementalLayoutEngineService(INodeIncrementalLayoutEngineRegistry registry) {
+        this.registry = Objects.requireNonNull(registry);
+    }
+
+    @Override
+    public Optional<INodeIncrementalLayoutEngine> getByNodeType(String nodeType) {
+        // @formatter:off
+        return this.registry.getNodeIncrementalLayoutEngines().stream()
+                .filter(engine -> engine.canLayout(nodeType))
+                .findFirst();
+        // @formatter:on
+    }
+
```

Comment:
I think this class and the previous one can both be removed

Reply:
The service has been removed